.github/workflows/workflow.yml

```diff
@@ -52,3 +52,45 @@ jobs:
 
     - name: Build & Test
       run: npm run test
+
+  test-proxy:
+    name: Test on ubuntu-latest
+    runs-on: ubuntu-latest
+    services:
+      squid:
+        image: dakale/squid
+        ports: ['3128:3128']
+        options: '--health-cmd "exit 0" --health-interval 3s'
+    env:
+      http_proxy: http://localhost:3128
+      https_proxy: http://localhost:3128
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+
+    - name: Get npm cache directory
+      id: npm-cache
+      run: |
+        echo "::set-output name=dir::$(npm config get cache)"
+
+    - uses: ./
+      with:
+        path: ${{ steps.npm-cache.outputs.dir }}
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Install
+      run: npm ci
+
+    - name: Prettier Format Check
```

Comment:
these next three steps shouldn't be here - the job above is about making sure build, unit, lint is covered.  This job is about validating the cache action works through a proxy and the steps above do that.  Notice setup-node pattern for this.

Reply:
Im not really sure the npm ci step after the cache restore is super useful either. All that would really be doing is making sure the build can still run after attempting to restore the npm cache from the cache service, which it may or may not, I dont really know the mechanism there. At least with installing and rerunning the tests you get a bit of insight into whether or not the cache action somehow corrupted the actual contents of the packages its restoring. 

I removed them anyways because I think a longer term solution wont use them but I think some future work would need to be planned to anyways to meet that outcome